src/_pytest/pytester.py

```diff
@@ -1266,15 +1306,15 @@ def run(
         __tracebackhide__ = True
 
         cmdargs = tuple(
-            str(arg) if isinstance(arg, py.path.local) else arg for arg in cmdargs
+            str(arg) if isinstance(arg, (py.path.local, Path)) else arg
```

Comment:
Should use `os.fspath` here instead of the `isinstance` (else it will be a breaking change when `py.path.local` is removed).